pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -285,29 +294,24 @@ func (r *Builder) Secret(vmRef ref.Ref, in, object *core.Secret) (err error) {
 	if err != nil {
 		return
 	}
-	thumbprint := string(in.Data["thumbprint"])
 	if hostDef, found := r.hosts[hostID]; found {
-		hostSecret, nErr := r.hostSecret(hostDef)
-		if nErr != nil {
-			err = nErr
-			return
+		if in, err := r.hostSecret(hostDef); err == nil {
+			object.Data = map[string][]byte{
```

Comment:
I think this can be simplified by moving this up:
```go
object.Data = map[string][]byte{
	"accessKeyId": in.Data["user"],
	"secretKey":   in.Data["password"],
}
```

then we can only check for the error

Reply:
but when there's a host specification defined, we want to get the user and password from there (it's not the same `in` secret)

Reply:
ack, missed the new declaration